rules-tests/DowngradePhp72/Rector/FuncCall/DowngradeJsonDecodeNullAssociativeArgRector/Fixture/clue_json_decode.php.inc

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Rector\Tests\DowngradePhp72\Rector\FuncCall\DowngradeJsonDecodeNullAssociativeArgRector\Fixture;
+
+/**
+ * @see \Clue\React\NDJson\Decoder
+ */
+final class ClueJsonDecode
+{
+    private $assoc;
+    private $depth;
+    private $options;
+
+    public function run($data)
+    {
+        if ($this->options === 0) {
+            $data = \json_decode($data, $this->assoc, $this->depth);
```

Comment:
@samsonasik The was that `mixed` type created infinite loop of adding conditions above.
For this we need the refresh scope feature.

Reply:
On decoder class, the `$this->assoc` is always filled in `__construct`, so that's seems not pointed the original behaviour, also, tried apply:

```
bin/rector p vendor/clue/ndjson-react/src/Decoder.php -c build/config/config-downgrade-parallel.php -n --clear-cache --debug 
```

and it will cause change multiple times, which shouldnot.

Reply:
I see, the variable should be skipped in case of set in constructor. I've added it to the test fixtures :+1: